<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SMEditorWindow</class>
 <widget class="QMainWindow" name="SMEditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cyberiada State Machine Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="smeditor.qrc">
    <normaloff>:/Icons/images/cyberiada.png</normaloff>:/Icons/images/cyberiada.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QSplitter" name="vSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QSplitter" name="hSplitter">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="CyberiadaSMView" name="SMView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="animated">
         <bool>true</bool>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
       </widget>
       <widget class="CyberiadaSMPropertiesWidget" name="propertiesWidget" native="true"/>
      </widget>
      <widget class="CyberiadaSMGraphicsView" name="sceneView"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionFont"/>
    <addaction name="separator"/>
    <addaction name="zoomInAction"/>
    <addaction name="zoomOutAction"/>
    <addaction name="fitContentAction"/>
    <addaction name="separator"/>
    <addaction name="transitionTextAction"/>
    <addaction name="inspectorModeAction"/>
    <addaction name="separator"/>
    <addaction name="propertiesAction"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="zoomInAction"/>
   <addaction name="zoomOutAction"/>
   <addaction name="fitContentAction"/>
   <addaction name="separator"/>
   <addaction name="selectToolAction"/>
   <addaction name="panAction"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionFont">
   <property name="text">
    <string>Font</string>
   </property>
  </action>
  <action name="zoomInAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/zoom-in.png</normaloff>:/Icons/images/zoom-in.png</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="zoomOutAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/zoom-out.png</normaloff>:/Icons/images/zoom-out.png</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="fitContentAction">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/fit-content.png</normaloff>:/Icons/images/fit-content.png</iconset>
   </property>
   <property name="text">
    <string>Fit Content</string>
   </property>
  </action>
  <action name="panAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/hand.png</normaloff>:/Icons/images/hand.png</iconset>
   </property>
   <property name="text">
    <string>Pan</string>
   </property>
   <property name="toolTip">
    <string>Pan</string>
   </property>
  </action>
  <action name="selectToolAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/cursor.png</normaloff>:/Icons/images/cursor.png</iconset>
   </property>
   <property name="text">
    <string>Select Tool</string>
   </property>
   <property name="toolTip">
    <string>Select Tool</string>
   </property>
  </action>
  <action name="transitionTextAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Текст на переходах</string>
   </property>
  </action>
  <action name="inspectorModeAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Режим инспектора</string>
   </property>
  </action>
  <action name="propertiesAction">
   <property name="text">
    <string>Настройки</string>
   </property>
  </action>
  <action name="actionDeleteElement">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/remove.png</normaloff>:/Icons/images/remove.png</iconset>
   </property>
   <property name="text">
    <string>Delete Element</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>&amp;Save as...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CyberiadaSMView</class>
   <extends>QTreeView</extends>
   <header>cyberiadasm_view.h</header>
   <slots>
    <signal>currentIndexActivated(QModelIndex)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>CyberiadaSMPropertiesWidget</class>
   <extends>QWidget</extends>
   <header>cyberiadasm_properties_widget.h</header>
   <container>1</container>
   <slots>
    <slot>slotElementSelected(QModelIndex)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>CyberiadaSMGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>cyberiadasm_editor_view.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="smeditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SMView</sender>
   <signal>currentIndexActivated(QModelIndex)</signal>
   <receiver>propertiesWidget</receiver>
   <slot>slotElementSelected(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>511</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>582</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAs</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotFileOpen()</slot>
  <slot>slotFileSave()</slot>
  <slot>slotFileSaveAs()</slot>
 </slots>
</ui>
